statsmodels/graphics/agreement.py

```diff
@@ -114,7 +114,7 @@ def mean_diff_plot(m1, m2, sd_limit=1.96, ax=None, scatter_kwds=None,
     ax.axhline(mean_diff, **mean_line_kwds)  # draw mean line.
 
     # Annotate mean line with mean difference.
-    ax.annotate(f'mean diff:\n{np.round(mean_diff, 2)}',
```

Comment:
I think you need to round first and then apply the format string, since I'm not sure the format string actually rounds - I think it might truncate, 

```python
 f"{4.229:0.2g}"
Out[2]: '4.2'

In [3]: f"{4.229:0.2g}"
Out[3]: '4.2'

In [5]: import numpy as np

In [6]: f"{np.round(4.229,2)}"
Out[6]: '4.23'
```

See how these are different.  You should ensure that they are the same.  I think the format should also be .3g ,
```
In [12]: f"{np.round(4.209,2)}"
Out[12]: '4.21'

In [13]: f"{np.round(4.209,2):.2g}"
Out[13]: '4.2'

In [14]: f"{np.round(4.209,2):.3g}"
Out[14]: '4.21'
```